101-perforce-helix-core-server/azuredeploy.json

```diff
@@ -0,0 +1,357 @@
+    },
+    "OS": {
+      "type": "string",
+      "allowedValues": ["Ubuntu 18.04 LTS", "CentOS 7.5", "RHEL 7.x"],
+      "metadata": {
+        "description": "The operating system of the VM."
+      },
+      "defaultValue": "CentOS 7.5"
+    },
+    "VMSize": {
+      "type": "string",
+      "metadata": {
+        "description": "Please select the size of the VM you wish to deploy.  Read more about sizing options here: https://docs.microsoft.com/en-us/azure/virtual-machines/linux/sizes-general"
+      },
+      "defaultValue": "Standard_B2s"
+    },
+    "dataDiskSize": {
+      "type": "int",
+      "minValue": 10,
+      "maxValue": 2048,
+      "metadata": {
+        "description": "Please select the size of the data disk you wish to deploy (value is integer GB) to hold your metadata, logs and depot (archive files). This can be any value up to 2TB (2048 GB)."
+      },
+      "defaultValue": 50
+    },
+    "p4Port": {
+      "type": "int",
+      "metadata": {
+        "description": "P4PORT value to connect to server (via ssl)"
+      }
+    },
+    "swarmPort": {
+      "type": "int",
+      "metadata": {
+        "description": "Port for Swarm (Apache) to be setup on."
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Admin username for Virtual Machine"
+      }
+    },
+    "adminSSHPubKey": {
+      "type": "securestring",
+      "metadata": {
+        "description": "SSH Public Key for the Virtual Machine."
+      }
+    },
+    "source_CIDR": {
+      "type": "string",
+      "metadata": {
+        "description": "CIDR block for SSH source - limit to your IP for secure access."
+      }
+    },
+    "helix_admin_password": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Helix Core Server superuser password (user perforce)."
+      }
+    }
+  },
+  "variables": {
+    "virtualNetworkName": "HXVNET",
+    "NSGName": "HXNSG",
+    "publicIPAddressType": "Dynamic",
+    "addressPrefix": "10.0.0.0/16",
+    "subnet1Name": "Subnet-1",
+    "subnet1Prefix": "10.0.0.0/24",
+    "numberOfInstances": 1,
+    "linuxscripturi": "[uri(parameters('_artifactsLocation'), concat('scripts/configure-linux.sh', parameters('_artifactsLocationSasToken')))]",
+    "virtualMachineSize": "[parameters('VMSize')]",
+    "imageReference": {
+      "Ubuntu 18.04 LTS": {
+        "publisher": "Canonical",
+        "offer": "UbuntuServer",
+        "sku": "18.04-LTS",
+        "version": "latest"
+      },
+      "CentOS 7.5": {
+        "publisher": "OpenLogic",
+        "offer": "CentOS",
+        "sku": "7.5",
+        "version": "latest"
+      },
+      "RHEL 7.x": {
+        "publisher": "RedHat",
+        "offer": "RHEL",
+        "sku": "7-LVM",
+        "version": "latest"
+      }
+    },
+    "dataDiskSize": "[parameters('dataDiskSize')]",
+    "publicIpName": "[substring(concat(uniqueString(resourceGroup().id, deployment().name)), 0, 6)]",
```

Comment:
make this idempotent (e.g. remove deployment().name) - also remove extraneous concat()